docs/guideline.md

```diff
@@ -43,6 +43,23 @@ type RPC struct {
 
 ## Tests
 
+Tests should be named according to their specific scope of testing. For
+instance, a unit test of a function should follow that convention:
+
+```go
+// dummy.go
+
+type Dummy struct{}
+
+func (d Dummy) String() string {...}
+```
+
+```go
+// dummy_test.go
+
+func TestDummy_String(t *testing.T) {}
+```
+
```

Comment:
I would add something to handle more complex situation, like:

For some situation, it is necessary to add detailed information about a test case. This can be done between the SCOPE and the FUNCTION:

```go
func Test_RPC_MultipleSimple_Stream(t *testing.T) {
...
}
```

As an invariant, the SCOPE and FUNCTION should always be the first, respectively the last, element.

Reply:
Done